src/components/NotificationProvider/messageBuilder.tsx

```diff
@@ -0,0 +1,49 @@
+import {
+  NotificationAction,
+  NotificationType,
+  QueuedNotification,
+} from "./types";
+import React, { ReactNode } from "react";
+
+export const queue = (notification: NotificationType): QueuedNotification => {
+  return { state: { queuedNotification: notification } };
+};
+
+export const info = (
+  message: string | ReactNode,
+  title?: string
+): NotificationType => {
+  return {
+    message,
+    title,
+    type: "information",
+  };
+};
+
+export const success = (message: string | ReactNode): NotificationType => {
+  return {
+    message,
+    type: "positive",
+  };
+};
+
+export const failure = (
+  title: string,
+  error: unknown,
+  message?: string | ReactNode,
+  actions?: NotificationAction[]
+): NotificationType => {
+  return {
+    actions,
+    message:
+      error && typeof error === "object" && "message" in error ? (
+        <>
+          {message} {error.message}
+        </>
```

Comment:
We're forcing the `error` object to have some very specific "shape" here in order to display a detailed error message. Maybe we should define a type for it at this point, and remove the `unknown`?

Reply:
When using `.catch((error) => ...` there is often no type information on the error available. We want to allow forwarding the error directly in that cases, hence we type it as unkown here, to allow `.catch((error) => notify.failure("some message", error))`

Reply:
My point was that currently this only works when the error object that we pass has this form:
```
{
  "message": "Something"
}
```
Which is unlikely to always be the case. I think this implementation is not robust and should be improved. We should not rely on the "message" key to be there to display a detailed error message. The key might be called "msg" or whatnot and then this would not work.

Reply:
The JS `Error` type has the message field, so when fetching a `throw new Error("foo")`, the code will pick up the right message. JS does not have proper type checking, and you can throw any object. But I wouldn't want to support (or know how to render) custom objects beeing thrown. I think expecting Errors (or the message field to be present) is a useful behavior.

Reply:
Ok, I get your motivations. So maybe this can be simplified a bit like this?
```suggestion
      error && error instanceof Error ? (
        <>
          {message} {error.message}
        </>
```
Since the motivation behind is "we assume the `error` to be an `Error` type, with a `message` field in it", right?